atlp-rwanda/ecommerce-app-legends-bn
# Ch tests setup

# chore[TESTS SETUP]  

 ## this is a chore branch for testing all endpoints and apis from this project, target is to get 100% of passing tests

 ## material(packages) used are :

 - jest and supertest for running tests.
 - jest-coverage-badge to show code coverage percentage.
 - 
 ## what I did:

- I have installed all packages for testing all apis, some of them are mentioned above.
 - I have created `jest.config.js` within root folder for making jest to use test database `elephantSQL`.
 - I have creates `jest.setup.js` for running migration and creating and managing collections used i
PS C:\Users\mwana\Documents\Andela\project\ecommerce-app-legends-bn> git commit -m " # chore[TESTS SETUP]

 ## this is a chore branch for testing all endpoints and apis from this project, target is to get 100% of passing tests

 ## material(packages) used are : 

 - `jest `and `supertest `for running tests.
 - `jest-coverage-badge` to show code coverage percentage.
 
 ## what I did: 

 - I have installed all packegs for testing all apis, some of them are mentioned above.
 - I have created jest.config.js within root folder for making jest to use test database ecomerce_database-test.
 - I have creates jest.setup.js for running migration and creating and managing collections used in testing.
 - Under a test folder I have created a users.test.js for tesing all user endpoints .
 - I have also added a coverage badge to a readme file which is counting around 97%.

 ## how to run:

- `git clone` from repo
 - run git checkout `ch-test-setup`
 - run `npm i`.
 - configure and create and configure database for tesing purpose (under test environment).       
 - run `npm test`.
 - to update badge run `npx upadate-badge`
 - check in your editor , terminal , and database and provide me a feedback. 

 # [WIP TESTS SETUP]"
